cupyx/scipy/interpolate/_bspline.py

```diff
@@ -0,0 +1,918 @@
+        for(n = 0; n < k + 1; n++) {
+            out[num_c * idx + j] = (
+                out[num_c * idx + j] +
+                c[(interval + n - k) * num_c + j] * h[n]);
+        }
+    }
+
+}
+'''
+
+D_BOOR_MODULE = cupy.RawModule(
+    code=D_BOOR_KERNEL, options=('-std=c++11',),
+    name_expressions=[f'd_boor<{type_name}>' for type_name in TYPES])
+
+
+DESIGN_MAT_KERNEL = r'''
+#include <cupy/complex.cuh>
+
+template<typename T, typename U>
+__global__ void compute_design_matrix(
+        const int k, const long long* intervals, T* bspline_basis,
+        T* data, U* indices) {
+
+    int idx = blockDim.x * blockIdx.x + threadIdx.x;
+    long long interval = *&intervals[idx];
+
+    T* work = bspline_basis + idx * (2 * k + 1);
+
+    for(int j = 0; j <= k; j++) {
+        int m = (k + 1) * idx + j;
+        data[m] = work[j];
+        indices[m] = (U) (interval - k + j);
+    }
+}
+'''
+
+DESIGN_MAT_MODULE = cupy.RawModule(
+    code=DESIGN_MAT_KERNEL, options=('-std=c++11',),
+    name_expressions=[f'compute_design_matrix<{type_name}, {itype}>'
+                      for type_name, itype in product(TYPES, INT_TYPES)])
+
+
+def _get_module_func(module, func_name, *template_args):
+    args_dtypes = [get_typename(arg.dtype) for arg in template_args]
+    template = ', '.join(args_dtypes)
+    kernel = module.get_function(f'{func_name}<{template}>')
+    return kernel
+
+
+def _get_dtype(dtype):
+    """Return np.complex128 for complex dtypes, np.float64 otherwise."""
+    if cupy.issubdtype(dtype, cupy.complexfloating):
+        return cupy.complex_
+    else:
+        return cupy.float_
+
+
+def _as_float_array(x, check_finite=False):
+    """Convert the input into a C contiguous float array.
+    NB: Upcasts half- and single-precision floats to double precision.
+    """
+    x = cupy.ascontiguousarray(x)
+    dtyp = _get_dtype(x.dtype)
+    x = x.astype(dtyp, copy=False)
+    if check_finite and not cupy.isfinite(x).all():
+        raise ValueError("Array must not contain infs or nans.")
+    return x
+
+
+def _evaluate_spline(t, c, k, xp, nu, extrapolate, out):
+    """
+    Evaluate a spline in the B-spline basis.
+
+    Parameters
+    ----------
+    t : ndarray, shape (n+k+1)
+        knots
+    c : ndarray, shape (n, m)
+        B-spline coefficients
+    xp : ndarray, shape (s,)
+        Points to evaluate the spline at.
+    nu : int
+        Order of derivative to evaluate.
+    extrapolate : int, optional
+        Whether to extrapolate to ouf-of-bounds points, or to return NaNs.
+    out : ndarray, shape (s, m)
+        Computed values of the spline at each of the input points.
+        This argument is modified in-place.
+    """
+    n = t.shape[0] - k - 1
+    intervals = cupy.empty_like(xp, dtype=cupy.int_)
+
+    # Compute intervals for each value
+    interval_kernel = _get_module_func(INTERVAL_MODULE, 'find_interval', xp)
+    interval_kernel((max(1, xp.shape[0] // 128),), (min(128, xp.shape[0]),),
```

Comment:
We need two blocks for, say, 129 elements.

```suggestion
    interval_kernel(((xp.shape[0] + 128 - 1) // 128,), (min(128, xp.shape[0]),),
```